.github/workflows/php-lint.yml

```diff
@@ -0,0 +1,59 @@
+name: PHP Code Linting
+
+on:
+  push:
+    branches:
+      - trunk
+      - 'feature/**'
+      - 'release/**'
```

Comment:
@borkweb Any reason we wouldn't just lint for any branch? I'm not confident folks will follow the prefixed conventions here.

Reply:
I was matching what seems to be in place in a couple of other WordPress repos. I'm good with opening it up 👍